File: agoraquiz/Models/Game.swift
//
//  Game.swift
//  agoraquiz
//
//  Created by Ibercivis on 15/5/24.
//

import Foundation

struct Game: Codable {
    let id: Int
    let score: Int
    let status: String
    let gameType: String
    let timeLeft: Int
    let questions: [Question]
    
    enum CodingKeys: String, CodingKey {
        case id
        case score
        case status
        case gameType = "game_type"
        case timeLeft = "time_left"
        case questions
    }
}

File: android/app/src/main/java/com/ibercivis/agora/classes/Game.java
package com.ibercivis.agora.classes;

import java.util.List;

public class Game {
    private int id;
    private List<Question> questions;
    private int score;
    private String status;

    private int timeLeft;

    // Constructor, getters y setters

    public Game(int id, List<Question> questions, int score, String status) {
        this.id = id;
        this.questions = questions;
        this.score = score;
        this.status = status;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public List<Question> getQuestions() {
        return questions;
    }

    public void setQuestions(List<Question> questions) {
        this.questions = questions;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public int getTimeLeft() {
        return timeLeft;
    }

    public void setTimeLeft(int timeLeft) {
        this.timeLeft = timeLeft;
    }
}
